odoo/addons/base/models/res_partner.py

```diff
@@ -33,9 +33,32 @@
 def _lang_get(self):
     return self.env['res.lang'].get_installed()
 
+def timezone_offset(tz):
+    """Take the timezone in a string format and return the hour offset.
+
+    e.g.
+        'Africa/Cairo' -> '+0200'
+    """
+    utcnow = datetime.datetime.utcnow()
+    localized_datetime = pytz.timezone(tz).localize(utcnow)
+    return localized_datetime.strftime('%z')
+
+def format_timezone(tz):
+    """Format the given timezone to add the offset.
+
+    e.g.
+        'Africa/Cairo' -> '(UTC+02:00) Africa/Cairo'
+    """
+    offset = timezone_offset(tz)
```

Comment:
Can't you use standard tools instead ? Notably ``get_timezone_gmt`` from babel for example (or equivalent, see https://babel.pocoo.org/en/latest/api/dates.html ) ? A lot of things already exist, not sure if you implemented it manually because it was not working or because those thinks are a bit hidden...

Reply:
From what I remember @std-odoo suggested not using other external libraries for a simple string formatting

Reply:
Not sure which message you are referencing to (I didn't know this method).
But this method seems to require a datetime to work, not sure sure it's usable here

Reply:
I really think doing string-based manipulation is a really bad idea ... and I don't think you dug enough in existing library as you do not seem certain of it ^^ . Could you have a deeper look of what exists instead ?